thunder/transforms/cudagraph.py

```diff
@@ -230,7 +234,7 @@ def _can_fuse_node(n: Node):
                 fused_bsyms.extend(bsyms)
             else:
                 region = Region(producers, consumers, bsyms)
-                fusion_bsym: BoundSymbol = self.fuse(region, fusion_counter, num_static_inputs)
+                fusion_bsym: BoundSymbol = self.fuse(region, fusion_counter)
```

Comment:
we are missing `num_static_inputs` here for a reason? If we get several non-isomorphic graphs, we should probably think about how to handle this parameter better. But that's not relevant now...

Reply:
I do think that this is one of the things where the default transform is not ideal, but it is not something that was used before: the parameter was there with a default, but there was no way of providing it.

Reply:
It is used in the backward pass. We are not loosing it there, are we not?

Reply:
Ouch, right.

Reply:
But yes, the design of this parameter is so-so as it was not expected to have graph breaks...

Reply:
but which were the args covered by it?

Reply:
So it seems that these were the saved for backwards of (???). Is the assumption here that they were either static in the forward (parameters) or copied to the input area of the forward cuda graph?
 
But with the old code:
```
import torch, thunder
with torch.device("cuda"):
    m = torch.nn.Linear(2, 3)
    inp = torch.randn(1, 2, requires_grad=True)
jm = thunder.jit(m, use_cudagraphs=True)

res = jm(inp)
grads = torch.autograd.grad(res.sum(), (inp, *m.parameters()))
```
the forward has no cudagraph, so in that case, having the input as a fixed is not really correct. (admittedly, a corner case).

Reply:
I guess the happy part is that we will get the parameters as static anyways and we will have to look into the buffers for our own good...